ibis/backends/duckdb/__init__.py

```diff
@@ -973,86 +973,133 @@ def read_delta(
         self.con.register(table_name, delta_table.to_pyarrow_dataset())
         return self.table(table_name)
 
-    def list_tables(
+    def _list_tables(
         self,
         like: str | None = None,
         database: tuple[str, str] | str | None = None,
-        schema: str | None = None,
     ) -> list[str]:
-        """List tables and views.
+        """List physical tables."""
 
-        ::: {.callout-note}
-        ## Ibis does not use the word `schema` to refer to database hierarchy.
+        table_loc = self._warn_and_create_table_loc(database)
 
-        A collection of tables is referred to as a `database`.
-        A collection of `database` is referred to as a `catalog`.
+        catalog = table_loc.catalog or self.current_catalog
+        database = table_loc.db or self.current_database
 
-        These terms are mapped onto the corresponding features in each
-        backend (where available), regardless of whether the backend itself
-        uses the same terminology.
-        :::
+        col = "table_name"
+        sql = (
+            sg.select(col)
+            .from_(sg.table("tables", db="information_schema"))
+            .where(
+                C.table_catalog.eq(sge.convert(catalog)),
+                C.table_schema.eq(sge.convert(database)),
+                C.table_type.eq("BASE TABLE"),
+            )
+            .sql(self.name, pretty=True)
```

Comment:
Now that you have this pattern established, I think it might make sense to pull this sqlglot query into a helper function -- it's the same query for each of the `list_` methods you are adding except the predicates to the `where` call, so maybe each `list_` function just constructs a list of those predicates and everything else is a shared implementation.